app/src/app_common.cc

```diff
@@ -433,6 +418,27 @@ void RegisterLibrariesFromUserAgent(const char* user_agent) {
   if (changed) registry->UpdateUserAgent();
 }
 
+void RegisterSdkUsage(void* platform_resource) {
+  MutexLock lock(*g_app_mutex);
+
+  // Only register libraries when no C++ apps was created before.
+  if (g_apps == nullptr) {
```

Comment:
This might be a nitpick, but even if g_apps currently correlates to whether or not an app has been created previously, I think it might be a bit safer if you added an IsInitialized() method to LibraryRegistry and check that.

Reply:
Agreed, this should probably check something else, like how on the Unity side there is a bool tracking registration.  If you do want to check if g_apps is null, either an assert or logging a warning/error seems like the better approach.

Reply:
Done.
I'm still using `g_app_mutex` to guard from race condition. Using the same mutex also has less chance for deadlock.